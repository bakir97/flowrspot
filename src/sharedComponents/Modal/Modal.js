import React from "react";
import styles from "./Modal.module.scss";
const Modal = ({
  children,
  modalType,
  closeModal,
  modalText,
  close,
  bigger
}) => {
  const handleClose = () => {
    closeModal();
    window.document.body.style.overflow = "auto";
  };
  const w =
    window.innerWidth ||
    document.documentElement.clientWidth ||
    document.body.clientWidth;
  const isMobile = w <= 760 ? true : false;
  if (modalType) {
    return (
      <div onClick={handleClose} className={styles.modal_background}>
        <div
          onClick={e => {
            e.stopPropagation();
          }}
          className={bigger ? styles.modal_profile : styles.modal}
        >
          {close && !isMobile && (
            <div className={styles.close_container}>
              <p onClick={handleClose} className={styles.close}>
                &#10005;
              </p>
            </div>
          )}
          {children}
        </div>
        <p onClick={handleClose} className={styles.modal_text}>
          {modalText}
        </p>
      </div>
    );
  }
  return null;
};

export default Modal;
