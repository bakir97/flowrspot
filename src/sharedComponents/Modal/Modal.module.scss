.modal_background {
  width: 100vw;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.5);
  height: 100%;
  top: 0;
  z-index: 1000;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  @media screen and (max-width: 479px) {
    background-color: white;
    top: 80px;
  }
}
.modal {
  border-radius: 3px;
  padding: 30px;
  width: 380px;
  box-shadow: 0 15px 30px 0 rgba(0, 0, 0, 0.05);
  background-color: white;
  @media screen and (max-width: 479px) {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
  }
}
.modal_profile {
  border-radius: 3px;
  width: 600px;
  box-shadow: 0 15px 30px 0 rgba(0, 0, 0, 0.05);
  background-color: white;
  @media screen and (max-width: 479px) {
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}
.modal_text {
  margin-top: 20px;
  color: white;
  opacity: 0.5;
  &:hover {
    cursor: pointer;
  }
  @media screen and (max-width: 479px) {
    display: none;
  }
}
.close_container {
  display: flex;
  .close {
    margin: 0;
    margin-right: 15px;
    margin-top: 15px;
    margin-bottom: 29px;
    margin-left: auto;
    color: black;
    text-align: end;
    &:hover {
      cursor: pointer;
    }
  }
}
