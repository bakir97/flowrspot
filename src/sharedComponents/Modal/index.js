import Modal from "./Modal";
import { connect } from "react-redux";
import { getModalType } from "../../redux/modal/selectors";
import { closeModal } from "../../redux/modal/actions";
const mapStateToProps = state => ({
  modalType: getModalType(state)
});

const mapDispatchToProps = {
  closeModal
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Modal);
