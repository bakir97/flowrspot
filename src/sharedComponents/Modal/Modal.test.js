import React from "react";
import { shallow } from "enzyme";
import Modal from "./Modal";
describe("Modal", () => {
  it("should render correctly", () => {
    const component = shallow(<Modal modalType />);
    expect(component).toMatchSnapshot();
  });
});
