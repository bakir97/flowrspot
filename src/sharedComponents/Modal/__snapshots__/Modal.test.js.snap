// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<div
  className="modal_background"
  onClick={[Function]}
>
  <div
    className="modal"
    onClick={[Function]}
  />
  <p
    className="modal_text"
    onClick={[Function]}
  />
</div>
`;
